sdk/servicebus/azure-servicebus/azure/servicebus/_common/auto_lock_renewer.py

```diff
@@ -120,7 +159,26 @@ def _renewable(self, renewable):
             return False
         return True
 
-    def _auto_lock_renew(
+    def _dispatch_worker(self):
+        self._last_activity_timestamp = time.time()
+        while not self._shutdown.is_set() and self._running.is_set():
+            while not self._renew_tasks.empty():
+                renew_task = self._renew_tasks.get()
+                if self._is_max_workers_greater_than_one:
+                    self._executor.submit(self._auto_lock_renew_task, *renew_task)
+                else:
+                    self._auto_lock_renew_task(*renew_task)
+                self._renew_tasks.task_done()
+                self._last_activity_timestamp = time.time()
+            # If there's no activity in the past self._idle_timeout seconds, exit the method
+            # This ensures the dispatching thread could exit, not blocking the main python thread
+            # the main worker thread could be started again if new tasks get registered
+            if time.time() - self._last_activity_timestamp >= self._idle_timeout:
+                self._running.clear()
+                self._last_activity_timestamp = None
```

Comment:
This shuts down the renew thread after a period of inactivity? That seems like a big change in behaviour?

Reply:
Nevermind - I actually read the comment. 
Why would the dispatching thread be blocking the main Python thread?

Reply:
as oob discussion, this is to handle the case that error being raised in the main python thread -- in such case if errors are not caught, then code after error raised (e.g. shutdown the renewer) can not be reached, so that the dispatcher is already running.

The idle timeout gives a way to let the dispatcher thread exit the loop